Core/src/Surfaces/EllipseBounds.cpp

```diff
@@ -44,31 +43,19 @@ void EllipseBounds::checkConsistency() noexcept(false) {
   }
 }
 
-bool EllipseBounds::inside(const Vector2& lposition,
-                           const BoundaryTolerance& boundaryTolerance) const {
-  if (boundaryTolerance.isInfinite()) {
-    return true;
-  }
-
-  if (auto absoluteBound = boundaryTolerance.asAbsoluteBoundOpt();
-      absoluteBound.has_value()) {
-    double tol0 = absoluteBound->tolerance0;
-    double tol1 = absoluteBound->tolerance1;
-
-    double phi =
-        detail::radian_sym(VectorHelpers::phi(lposition) - get(eAveragePhi));
-    double phiHalf = get(eHalfPhiSector) + tol1;
-
-    bool insidePhi = (-phiHalf <= phi) && (phi < phiHalf);
-    bool insideInner = (get(eInnerRx) <= tol0) || (get(eOuterRx) <= tol0) ||
-                       (1 < (square(lposition[0] / (get(eInnerRx) - tol0)) +
-                             square(lposition[1] / (get(eOuterRx) - tol0))));
-    bool insideOuter = (square(lposition[0] / (get(eInnerRy) + tol0)) +
-                        square(lposition[1] / (get(eOuterRy) + tol0))) < 1;
-    return insidePhi && insideInner && insideOuter;
-  }
+bool EllipseBounds::inside(const Vector2& lposition) const {
+  double phi =
+      detail::radian_sym(VectorHelpers::phi(lposition) - get(eAveragePhi));
+  return (-get(eHalfPhiSector) <= phi) && (phi < get(eHalfPhiSector)) &&
+         (square(lposition[eBoundLoc0] / get(eInnerRx)) +
+          square(lposition[eBoundLoc1] / get(eInnerRy))) >= 1 &&
+         (square(lposition[eBoundLoc0] / get(eOuterRx)) +
+          square(lposition[eBoundLoc1] / get(eOuterRy))) < 1;
+}
 
-  throw std::logic_error("Unsupported boundary check type");
+Vector2 EllipseBounds::closestPoint(const Vector2& /*lposition*/,
+                                    const SquareMatrix2& /*metric*/) const {
+  throw std::logic_error("Not implemented");
```

Comment:
Do we need it?

Reply:
I think until now we did not come across it but makes sense to implement it. Should not be too hard